### PR TITLE
Add CI check for uncommitted migrations

### LINKED ISSUE #571: Run `makemigrations` in CI to prevent merging PRs with missing migrations.
## Description

It's currently possible to merge PRs with missing migrations.

## Reproduction

1.  See #570 

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Fixes #571  by @sarayourfriend 

## Description

Adds a new check to verify there are no uncommitted migrations in the PR.

## Testing Instructions

Verify that the test commit (43c8c2c581b7d44a2a80829d0fb0782cae2ddd57) introduces uncommitted migrations and that the CI fails for that commit.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [N/A] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [N/A] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: justfile [hunk 1]
```diff
@@ -165,17 +165,21 @@ _api-install:
     "Waiting for the API to be healthy..."
 
 # Run API tests inside Docker
-api-test docker_args="" tests="": up wait-for-es wait-for-ing wait-for-web
-    docker-compose exec {{ docker_args }} web ./test/run_test.sh {{ tests }}
+@api-test docker_args="" tests="":
+    docker-compose exec {{ docker_args }} ./test/run_test.sh {{ tests }}
```
Comment (AetherUnbound) [inline justfile:169]: Were `web` and the other dependencies intended to be removed here?

File: justfile [hunk 2]
```diff
@@ -165,17 +165,21 @@ _api-install:
...
 
 # Run API tests locally
 api-testlocal args="":
     cd api && pipenv run ./test/run_test.sh {{ args }}
 
-# Run Django administrative commands
-dj args="":
+# Run Django administrative commands locally
+dj-local +args:
```
Comment (AetherUnbound) [inline justfile:176]: Love this!

**Review by AetherUnbound — COMMENTED** [review]
It looks like the [most recent CI run](https://github.com/WordPress/openverse-api/runs/5649770842?check_suite_focus=true) failed because it was unable to run the `just` command, not because of an uncommitted migration :thinking:

File: justfile (hunk 1 above)
Comment (sarayourfriend) [reply justfile:169]: No

Comment (sarayourfriend) [reply justfile:169]: I was trying to deduplicate the dependencies list; alas, the `docker_args` argument is a pain in the butt to try to pass through empty.

**Comment by sarayourfriend** [comment]
> It looks like the [most recent CI run](https://github.com/WordPress/openverse-api/runs/5649770842?check_suite_focus=true) failed because it was unable to run the just command, not because of an uncommitted migration thinking

Yes, so far it still hasn't run correctly. I only undrafted this so I could get the CI to run, please ignore for now!

**Comment by AetherUnbound** [comment]
> Yes, so far it still hasn't run correctly. I only undrafted this so I could get the CI to run, please ignore for now!

Oh got it, my bad! I didn't realize/remember CI doesn't run on drafts

**Comment by sarayourfriend** [comment]
Seems like the TTY value in the docker-compose won't work for CI :thinking: Why does it work for running the tests?

**Comment by sarayourfriend** [comment]
This run shows it working to pass when there are no uncommitted migrations: https://github.com/WordPress/openverse-api/runs/5680076354

The run was a mistake but at least it proves that it works :slightly_smiling_face: 

The next run will show what happens when it should break.

File: justfile [hunk 3]
```diff
@@ -164,8 +164,11 @@ _api-install:
     '"$(just web-health)" != "200"' \
     "Waiting for the API to be healthy..."
 
+@_api-up: up wait-for-es wait-for-ing wait-for-web
```
Comment (AetherUnbound) [inline justfile:167]: Smart!

**Comment by sarayourfriend** [comment]
@AetherUnbound @obulat @stacimc this is ready for review now.

You can see that the workflow correctly fails when there are uncommitted migrations: https://github.com/WordPress/openverse-api/runs/5680410997?check_suite_focus=true

And passes when a new migration is added: https://github.com/WordPress/openverse-api/runs/5680076354

File: justfile [hunk 4]
```diff
@@ -164,18 +164,25 @@ _api-install:
...
     cd api && pipenv run ./test/run_test.sh {{ args }}
 
-# Run Django administrative commands
-dj args="":
+# Run Django administrative commands locally
+dj-local +args:
     cd api && pipenv run python manage.py {{ args }}
 
+# Run Django administrative commands in the docker container
+@dj docker_args="" +args="": _api-up
```
Comment (AetherUnbound) [inline justfile:183]: Thank you for this!!

**Review by AetherUnbound — APPROVED** [review]
This is awesome :tada: great work!

**Review by stacimc — APPROVED** [review]
🎉 Excited to see this go in!

**Pull request merged**